TS18028: Private identifiers error when compiling puppeteer-core types with TypeScript

<img width="1371" height="319" alt="Image" src="https://github.com/user-attachments/assets/9363020c-8a3f-4a88-9d2c-d7b2e47438ef" />

When using puppeteer-core in a TypeScript project, compilation fails with the following error:

```
node_modules/puppeteer-core/lib/types.d.ts:33:5 - error TS18028: Private identifiers are only available when targeting ECMAScript 2015 and higher.

33     #private;
```


I simply  keep default the tscofing.json file that is initialy generate by running command f
`npx tsc --init`

this is my tsconfig.json file 
```
{
  // Visit https://aka.ms/tsconfig to read more about this file
  "compilerOptions": {
    // File Layout
    // "rootDir": "./src",
    // "outDir": "./dist",

    // Environment Settings
    // See also https://aka.ms/tsconfig/module
    "module": "nodenext",
    "target": "esnext",
    "types": [],
    // For nodejs:
    // "lib": ["esnext"],
    // "types": ["node"],
    // and npm install -D @types/node

    // Other Outputs
    "sourceMap": true,
    "declaration": true,
    "declarationMap": true,

    // Stricter Typechecking Options
    "noUncheckedIndexedAccess": true,
    "exactOptionalPropertyTypes": true,

    // Style Options
    // "noImplicitReturns": true,
    // "noImplicitOverride": true,
    // "noUnusedLocals": true,
    // "noUnusedParameters": true,
    // "noFallthroughCasesInSwitch": true,
    // "noPropertyAccessFromIndexSignature": true,

    // Recommended Options
    "strict": true,
    "jsx": "react-jsx",
    "verbatimModuleSyntax": true,
    "isolatedModules": true,
    "noUncheckedSideEffectImports": true,
    "moduleDetection": "force",
    "skipLibCheck": true,
  }
}
```


Expected Behavior:
TypeScript should compile without errors or the package should document the required target version clearly.

Actual Behavior:
Compilation fails with TS18028 because #private fields in the type definitions require ECMAScript 2015 or higher.

I share my package.json file also
```
{

<img width="1371" height="319" alt="Image" src="https://github.com/user-attachments/assets/034f8275-992e-47f5-9ef8-efa046953076" />

  "type": "module",
  "main": "index.js",
  "scripts": {
    "build": "npx tsc",
    "dev": "node src/client.js",
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "dependencies": {
    "jimp": "^1.6.0",
    "jsqr": "^1.4.0",
    "node-fetch": "^3.3.2",
    "puppeteer": "^24.23.0",
    "qrcode-terminal": "^0.12.0",
    "winston": "^3.18.3"
  },
  "devDependencies": {
    "ts-node": "^10.9.2",
    "tsx": "^4.20.6"
  }
}


```

Enviroment
- Puppeteer : 24.23.0
- NodeJs : 22.0
- OS : Windows 11

Possible Solution / Workaround:
- Update tsconfig.json to use "target": "ES2015" or higher.

--- a/docs/guides/system-requirements.md
+++ b/docs/guides/system-requirements.md
@@ -4,7 +4,8 @@
   [maintenance LTS](https://github.com/nodejs/Release#release-schedule) version of
   Node
 
-- TypeScript 4.7.4+ (If used with TypeScript)
+- TypeScript 4.7.4+ (If used with TypeScript).
+  - Target ES2022 or later if you [type check node_modules](https://www.typescriptlang.org/tsconfig/#skipLibCheck).
 
 - Chrome for Testing browser system requirements:
   - [Windows](https://support.google.com/chrome/a/answer/7100626?hl=en#:~:text=the%20specified%20criteria.-,Windows,-To%20use%20Chrome), x64 architecture

